src/Hooks/GlobalMiddleware.php

```diff
@@ -29,11 +31,19 @@ public function __construct(
 
     public function handle(Request $request, Closure $next): mixed
     {
+        $this->nightwatch->state->executionPreview = Str::tinyText(
+            $request->getMethod().' '.$request->getBaseUrl().$request->getPathInfo()
+        );
```

Comment:
Some of these methods run code that throws exceptions. Although these particular code paths are very unlikely to throw, I would prefer we continue to catch any exception that could impact the execution of the consuming application.

This is our current pattern for handling these things.

```suggestion
        try {
          $this->nightwatch->state->executionPreview = Str::tinyText(
              $request->getMethod().' '.$request->getBaseUrl().$request->getPathInfo()
          );
      } catch (Throwable $e) {
          $this->nightwatch->report($e);
      }
```

The `nightwatch->report` method is "exception safe", i.e., it will never throw unless a fatal exception occurs, .e.g, out of memory.

Reply:
Good call!